e2e/test/scenarios/embedding/sdk-iframe-embedding/sdk-iframe-embedding.cy.spec.ts

```diff
@@ -246,6 +246,35 @@ describe("scenarios > embedding > sdk iframe embedding", () => {
     cy.log("3. dashboard title should now be visible");
     getIframeWindow().findByText("Orders in a dashboard").should("be.visible");
   });
+
+  it("CSP nonces are set for custom expression styles (EMB-707)", () => {
+    const frame = H.loadSdkIframeEmbedTestPage({ template: "exploration" });
+
+    frame.within(() => {
+      cy.findByText("Orders").should("be.visible");
+
+      H.popover().within(() => {
+        cy.findByText("Orders").click();
+      });
+
+      cy.log("csp nonces should be set");
+      cy.get("style[nonce]")
+        .should("have.length.greaterThan", 0)
+        .first()
+        .should("have.attr", "nonce")
+        .and("have.length.greaterThan", 4);
+
+      cy.findByRole("button", { name: "Custom column" }).click();
+
+      cy.log("injected codemirror styles should be set");
+      cy.findByTestId("custom-expression-query-editor")
+        .should("be.visible")
+        .find(".cm-editor .cm-placeholder")
+        .and("have.css", "color", "rgb(136, 136, 136)");
```

Comment:
Without the CSP nonce, this placeholder color won't be set as CodeMirror uses https://www.npmjs.com/package/style-mod to set styles. Just a sanity check.

<img width="3572" height="2046" alt="image" src="https://github.com/user-attachments/assets/98da3955-ec60-4089-8d52-abad4c8df34e" />